Gems/ROS2/Code/Source/RobotImporter/Utils/SourceAssetsStorage.cpp

```diff
@@ -371,8 +371,11 @@ namespace ROS2::Utils
                     resolvedPath.c_str(),
                     targetPathAssetTmp.c_str(),
                     outcomeCopyTmp.GetResultCode());
+
                 if (outcomeCopyTmp)
                 {
+                    FlushIOOfAsset(targetPathAssetTmp);
```

Comment:
```suggestion
                    // An I/O flush is required here because we need to load the Scene file into memory from the temporary directory
                    // to generate a proper manifest for it in the final directory before the Scene file itself is moved into the final
                    // directory and processed. This ensures that when the Scene file is detected by the Asset Processor in the final
                    // location, it will already have the desired export settings in the manifest to cause it to be exported correctly.
                    // If we didn't flush the I/O here, the Scene file load can fail because the load contains a call to 
                    // AssetSystemComponent::GetSourceInfoBySourcePath that will fail if the Asset Processor hasn't detected the
                    // existence of the file yet. With the flush, everything works correctly.

                    FlushIOOfAsset(targetPathAssetTmp);
```